Resources/public/js/views/ez-languageswitcherview.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-languageswitcherview', function (Y) {
+    "use strict";
+    /**
+     * Provides the Language Switcher View class
+     *
+     * @module ez-languageswitcherview
+     */
+    Y.namespace('eZ');
+
+    var EXPANDED_LANGUAGE_LIST = 'is-expanded';
+
+    /**
+     * The raw content view
+     *
+     * @namespace eZ
+     * @class LanguageSwitcherView
+     * @constructor
+     * @extends eZ.TemplateBasedView
+     */
+    Y.eZ.LanguageSwitcherView = Y.Base.create('languageSwitcherView', Y.eZ.TemplateBasedView, [Y.eZ.AccordionElement], {
+        events: {
+            '.ez-selected-language': {
+                'tap': '_expandList'
+            },
+            '.ez-content-translation a': {
+                'tap': '_switchLanguage'
+            }
+        },
+
+        initializer: function () {
+        },
+
+        /**
+         * Renders the language switcher view
+         *
+         * @method render
+         * @return {eZ.LanguageSwitcherView} the view itself
+         */
+        render: function () {
+            var container = this.get('container');
+
+            container.setHTML(this.template({
+                content: this.get('content').toJSON(),
+                //todo: get current translation and pass it here
+                currentTranslation: {'flagCode': 'gb', 'language': 'eng-GB', 'languageCode': 'eng-GB'},
+                translations: this._getOtherTranslationsList()
+            }));
+
+            return this;
+        },
+
+        /**
+         * Returns array with translations other than already being viewed
+         *
+         * @method _getOtherTranslationsList
+         * @private
+         * @return {Array} other translations
+         */
+        _getOtherTranslationsList: function () {
+            //todo: get list of existing translations for current content (miss current translation?)
+            var translations = [
+                    {'flagCode': 'de', 'language': 'ger-DE', 'languageCode': 'ger-DE'},
+                    {'flagCode': 'pl', 'language': 'pol-PL', 'languageCode': 'pol-PL'},
+                    {'flagCode': 'fr', 'language': 'fre-FR', 'languageCode': 'fre-FR'}
+                ];
+
+            return translations;
+        },
+
+        /**
+         * Exapands translations list
+         *
+         * @method _expandList
+         * @private
+         */
+        _expandList: function () {
+            var container = this.get('container');
+
+            container.one('.ez-expandable-area').toggleClass(EXPANDED_LANGUAGE_LIST);
+        },
+
+        _switchLanguage: function (e) {
+            var location = this.get('location'),
+                languageCode = e.target._node.getAttribute('data-language-code');
+
+            this.get('app').navigateTo(
```

Comment:
you don't need to do that, instead the language switcher view should generate the links (there's a `path` template function to generate proper links) so that we let the browser do the navigation.